File: src/tests/fidl/source_compatibility/enum-flexible-strict/rust/step_03_after.rs
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#![allow(dead_code)]

use fidl_fidl_test_enumflexiblestrict as fidl_lib;

// [START contents]
fn complement(color: &fidl_lib::Color) -> fidl_lib::Color {
    match color {
        fidl_lib::Color::Red => fidl_lib::Color::Blue,
        fidl_lib::Color::Blue => fidl_lib::Color::Red,
    }
}

// [END contents]

fn main() {}

File: src/tests/fidl/source_compatibility/enum-strict-flexible/llcpp/step_03_after.cc
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <fidl/test/enumstrictflexible/llcpp/fidl.h>  // nogncheck
namespace fidl_test = llcpp::fidl::test::enumstrictflexible;

// [START contents]
fidl_test::wire::Color writer(std::string s) {
  if (s == "red") {
    return fidl_test::wire::Color::RED;
  } else if (s == "blue") {
    return fidl_test::wire::Color::BLUE;
  } else {
    return fidl_test::wire::Color::Unknown();
  }
}

std::string reader(fidl_test::wire::Color color) {
  if (color.IsUnknown()) {
    return "unknown";
  }
  switch (color) {
    case fidl_test::wire::Color::RED:
      return "red";
    case fidl_test::wire::Color::BLUE:
      return "blue";
    default:
      return "error";
  }
}
// [END contents]

int main(int argc, const char** argv) { return 0; }
